documentation/config_file_puppetserver.md

```diff
@@ -0,0 +1,93 @@
+    * `client-whitelist` contains an array of client certificate names that are whitelisted to access the admin API. Puppet Server denies any requests made to this endpoint that do not present a valid client certificate mentioned in this array.
+
+    If neither the `authorization-required` nor the `client-whitelist` settings are specified, Puppet Server uses the new authorization methods and [`auth.conf` documentation][] formats to access the admin API endpoints.
+
+~~~
+# Configuration for the JRuby interpreters.
+
+jruby-puppet: {
+    ruby-load-path: [/opt/puppetlabs/puppet/lib/ruby/vendor_ruby]
+    gem-home: /opt/puppetlabs/server/data/puppetserver/jruby-gems
+    master-conf-dir: /etc/puppetlabs/puppet
+    master-code-dir: /etc/puppetlabs/code
+    master-var-dir: /opt/puppetlabs/server/data/puppetserver
+    master-run-dir: /var/run/puppetlabs/puppetserver
+    master-log-dir: /var/log/puppetlabs/puppetserver
+    max-active-instances: 1
+    max-requests-per-instance: 0
+    use-legacy-auth-conf: false
+}
+
+# Settings related to HTTP client requests made by Puppet Server.
+http-client: {
+    # A list of acceptable protocols for making HTTP requests
+    #ssl-protocols: [TLSv1, TLSv1.1, TLSv1.2]
+
+    # A list of acceptable cipher suites for making HTTP requests. For more info on available cipher suites, see:
+    # http://docs.oracle.com/javase/7/docs/technotes/guides/security/SunProviders.html#SunJSSEProvider
+    #cipher-suites: [TLS_RSA_WITH_AES_256_CBC_SHA256,
+    #                TLS_RSA_WITH_AES_256_CBC_SHA,
+    #                TLS_RSA_WITH_AES_128_CBC_SHA256,
+    #                TLS_RSA_WITH_AES_128_CBC_SHA]
+
+    # The amount of time, in milliseconds, that an outbound HTTP connection
+    # will wait for data to be available before closing the socket. If not
+    # defined, defaults to 20 minutes. If 0, the timeout is infinite and if
+    # negative, the value is undefined by the application and governed by the
+    # system default behavior.
+    #idle-timeout-milliseconds: 1200000
+
+    # The amount of time, in milliseconds, that an outbound HTTP connection will
+    # wait to connect before giving up. Defaults to 2 minutes if not set. If 0,
+    # the timeout is infinite and if negative, the value is undefined in the
+    # application and governed by the system default behavior.
+    #connect-timeout-milliseconds: 120000
+}
+
+# Settings related to profiling the puppet Ruby code.
+profiler: {
+    enabled: true
+}
+~~~
+
+> **Note:** The `puppet-admin` setting and `client-whitelist` parameter are deprecated in favor of new authorization methods provided in Puppet Server 2.2. For details, see the [`auth.conf` documentation][./config_file_auth.html].
```

Comment:
Actually, I did a few things wrong here: you're right that I could use the reference link, but the reference link at the top points to the wrong place (`conf_file_auth` sted `config_file_auth`). And those should've been parens instead of brackets for a local link anyway.